BeeKit/Managers/GoalManager.swift

```diff
@@ -25,6 +25,9 @@ public actor GoalManager {
     private nonisolated let currentUserManager: CurrentUserManager
 
     public var goalsFetchedAt : Date? = nil
+    
+    // id, not goalname (slug)
+    private var lastFetchedByGoalId: [String: Date?] = [:]
```

Comment:
I think `Goal.lastModifiedLocal` may already be tracking the value you want?

Reply:
I was looking for when the app last fetched the data from the server. With Core Data, theoretically we could be editing objects locally (offline mode?) and at some point syncing them with the server.

Given _local_ in the name and the description of the attribute,
https://github.com/beeminder/BeeSwift/blob/745cf9875ed6b0303c46d5d8e4c0a2d44796040d/BeeKit/Model/Goal.swift#L66-L67
I had the impression this is meant to reflect when the goal was last touched locally and does not necessarily reflect also when the goal was last fetched from the server. I see this date is being set whenever updateToMatch is being called which is being called after a fetch of user/goal from the backend.

Reply:
I see it, `goal.lastModifiedLocal` is also set with Date.now when the object is created (pulled from the db). In this case the lastModifiedLocal does not reflect how fresh the data is (when last pulled from server).

Reply:
> goal.lastModifiedLocal is also set with Date.now when the object is created (pulled from the db). 

I don't believe this is true, at the very least it's not the intended behavior. While there is a constructor which sets the value, I don't believe this is the constructor which is invoked when materializing from the database.

Reply:
Alright, you convinced me!

Reply:
Yeah, I thought the explicit list all attributes constructors might be useful for tests. If we had more tests.